### PR TITLE
Trigger to allow injecting view classes

### PR DESCRIPTION
**Pull request by carcam**
* Trigger onBeforeLoadView event to inject View Classes for Event Details
* Disable caching when pdf format (needed for PhocaPDF plugin)

### CODE REVIEW

File: component/site/controllers/icalevent.php [hunk 1]
```diff
@@ -107,7 +111,7 @@ function detail() {
 		// View caching logic -- simple... are we logged in?
 		$cfg	 = JEVConfig::getInstance();
 		$joomlaconf = JFactory::getConfig();
-		$useCache = intval($cfg->get('com_cache', 0)) && $joomlaconf->get('caching', 1);
+		$useCache = intval($cfg->get('com_cache', 0)) && $joomlaconf->get('caching', 1) && $viewType != "pdf";
 		$user = JFactory::getUser();
```
Comment (GeraintEdwards) [inline component/site/controllers/icalevent.php]: why not have a variable e.g. $useCache and change it in the plugin - then its not hard coded in here

Comment (GeraintEdwards) [reply component/site/controllers/icalevent.php]: move it up

**Comment by carcam** [comment]
Hi @GeraintEdwards, please check it now ;)

File: component/site/controllers/icalrepeat.php [hunk 2]
```diff
@@ -116,6 +116,10 @@ function detail() {
 		$theme = JEV_CommonFunctions::getJEventsViewName();
 
 		$view = "icalevent";
+
+		$dispatcher = JEventDispatcher::getInstance();
+		$dispatcher->trigger('onBeforeLoadView', array($view, $theme, $viewType, 'icalrepeat.detail'));
```
Comment (GeraintEdwards) [inline component/site/controllers/icalrepeat.php]: This needs $useCache to be passed in

**Comment by carcam** [comment]
Hi @GeraintEdwards this is fixed now.

Best!!

**Pull request merged**